Introduce support for JEP 483's new AOT cache (#46169)
quarkusio/quarkus: docs/src/main/asciidoc/appcds.adoc

AppCDS has improved significantly in the latest JDK releases. This means that to ensure the best possible improvements from it, make sure your projects targets the highest possible Java version (ideally 17 or 21).
====

[TIP]
====
Starting with JDK 24, the JVM provides an evolution of the class-data sharing in the form of the AOT cache.

> If you are on building an application that will target JDK 24+ you can take advantage of this feature by additionally using

Reviewer: ```suggestion
If you are building an application that will target JDK 24+ you can take advantage of this feature by adding the following system property when packaging your application:
```

> If you are on building an application that will target JDK 24+ you can take advantage of this feature by additiionally using

Reviewer 2: ```suggestion
If you are on building an application that will target JDK 24+ you can take advantage of this feature by additionally using
```

[source,bash]
----
-Dquarkus.package.jar.appcds.use-aot=true
----

The result of this flag (plus the `-Dquarkus.package.jar.appcds.use-aot=true` original one) is the creation of an AOT cache file
named `app.aot`.

> You can use when launching the application like so:

Reviewer: ```suggestion
You can use this AOT cache when launching the application like so:
```
